### PR TITLE
Use hasSameLength

### PR DESCRIPTION
**Pull request by dwijnand**

### CODE REVIEW

**Comment by TheElectronWill** [comment]
`hasSameLengthAs` uses `@tailrec`, I wonder if that makes a difference here 🤔

**Comment by TheElectronWill** [comment]
test performance please

**Review by TheElectronWill — APPROVED** [review]

File: compiler/src/dotty/tools/dotc/core/Decorators.scala [hunk 1]
```diff
@@ -184,6 +184,9 @@ object Decorators {
       loop(xs, xs, 0)
     end mapWithIndexConserve
 
+    /** True if two lists have the same length.  Since calling length on linear sequences
+     *  is O(n), it is an inadvisable way to test length equality.
```
Comment (nicolasstucki) [inline compiler/src/dotty/tools/dotc/core/Decorators.scala:188]: Maybe also mention that this is `O(n min m)`.

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/core/Decorators.scala:188]: Actually it is ` Θ(n min m)`

**Review by nicolasstucki — APPROVED** [review]
Otherwise LGTM

**Pull request merged**